internal/receiver/smartagentreceiver/config_test.go

```diff
@@ -416,3 +417,77 @@ func TestLoadInvalidConfigWithNonStringArrayDimensionClients(t *testing.T) {
 		"error reading receivers configuration for smartagent/haproxy: dimensionClients must be an array of compatible exporter names")
 	require.Nil(t, cfg)
 }
+
+func TestFilteringConfig(t *testing.T) {
+	factories, err := componenttest.ExampleComponents()
+	assert.Nil(t, err)
+
+	factory := NewFactory()
+	factories.Receivers[configmodels.Type(typeStr)] = factory
+	cfg, err := configtest.LoadConfigFile(
+		t, path.Join(".", "testdata", "filtering_config.yaml"), factories,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+
+	fsCfg := cfg.Receivers["smartagent/filesystems"].(*Config)
+	require.Equal(t, &Config{
+		ReceiverSettings: configmodels.ReceiverSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr + "/filesystems",
+		},
+		monitorConfig: &filesystems.Config{
+			MonitorConfig: config.MonitorConfig{
+				Type: "filesystems",
+				DatapointsToExclude: []config.MetricFilter{
+					{
+						MetricName: "df_inodes.*",
+						Dimensions: map[string]interface{}{
+							"mountpoint": []interface{}{"*", "!/hostfs/var/lib/cni"},
+						},
+					},
+				},
+				ExtraGroups:  []string{"inodes"},
+				ExtraMetrics: []string{"percent_bytes.reserved"},
+			},
+		},
+	}, fsCfg)
+	require.NoError(t, fsCfg.validate())
+}
+
+func TestInvalidFilteringConfig(t *testing.T) {
+	factories, err := componenttest.ExampleComponents()
+	assert.Nil(t, err)
+
+	factory := NewFactory()
+	factories.Receivers[configmodels.Type(typeStr)] = factory
+	cfg, err := configtest.LoadConfigFile(
+		t, path.Join(".", "testdata", "invalid_filtering_config.yaml"), factories,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+
+	fsCfg := cfg.Receivers["smartagent/filesystems"].(*Config)
+	require.Equal(t, &Config{
+		ReceiverSettings: configmodels.ReceiverSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr + "/filesystems",
+		},
+		monitorConfig: &filesystems.Config{
+			MonitorConfig: config.MonitorConfig{
+				Type: "filesystems",
+				DatapointsToExclude: []config.MetricFilter{
+					{
+						MetricNames: []string{"./[0-"},
+					},
+				},
+			},
+		},
+	}, fsCfg)
+
+	err = fsCfg.validate()
+	require.Error(t, err)
+	require.EqualError(t, err, "unexpected end of input")
```

Comment:
Not necessary to address here, but we should probably provide an error wrapper for validation errors so they provide a bit more context.  This is pretty vague.

Reply:
Makes sense, submitted an issue https://github.com/signalfx/splunk-otel-collector/issues/157. Will cover it in a following PR